src/main/java/hill/weatherconfidence/ingest/GeocodeClient.java

```diff
@@ -0,0 +1,31 @@
+package hill.weatherconfidence.ingest;
+
+import hill.weatherconfidence.ingest.model.GeocodeResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+public class GeocodeClient {
+
+    private static final Logger log = LoggerFactory.getLogger(GeocodeClient.class);
+
+    private final WebClient webClient;
+    private final String baseUrl;
+    private final String appId;
+
+    public GeocodeClient(String baseUrl, String appId) {
+        this.baseUrl = baseUrl;
+        this.appId = appId;
+        this.webClient = WebClient.create();
+    }
+
+    public Mono<GeocodeResponse> geocode(String location) {
```

Comment:
This is the bit that actually calls the web server for the geocoded data.  It uses reactive (non-blocking) code, although at the moment we're blocking to wait for the response in the calling code.